section `.text1' will not fit in region `iram1_0_seg'

### The problem

After upgrade to 2022.2.1 (from 2022.1.3) installation aborts with 
`
/data/cache/platformio/packages/toolchain-xtensa/bin/../lib/gcc/xtensa-lx106-elf/10.3.0/../../../../xtensa-lx106-elf/bin/ld: /data/room2-info/.pioenvs/room2-info/firmware.elf section `.text1' will not fit in region `iram1_0_seg'
collect2: error: ld returned 1 exit status
*** [/data/room2-info/.pioenvs/room2-info/firmware.elf] Error 1
`

With version 2022.1.3 everything worked. Also the delete of the old build files didn't helped.

I already tried to delete build files but no change.



### Which version of ESPHome has the issue?

2022.2.1

### What type of installation are you using?

Home Assistant Add-on

### Which version of Home Assistant has the issue?

2022.2.7

### What platform are you using?

ESP8266

### Board

d1_mini

### Component causing the issue

iram (perhaps)

### Example YAML snippet

```yaml
substitutions:
  devicename: room2-info
  friendly_name: Room2 Info
  device_description: Information from Room2

esphome:
  name: $devicename
  comment: ${device_description}
  platform: ESP8266
  board: d1_mini

wifi:
  ssid: "123"
  password: "123"

  # Enable fallback hotspot (captive portal) in case wifi connection fails
  ap:
    ssid: "${friendly_name} Fallback Hotspot"
    password: "123"

captive_portal:

# Enable logging
logger:

# Enable Home Assistant API
api:
  password: "123"

ota:
  password: "123"

improv_serial:


time:
  - platform: homeassistant
    id: esptime

sensor:
  - platform: uptime
    name: "${friendly_name} Uptime"
    entity_category: diagnostic
  - platform: wifi_signal
    name: "${friendly_name} WiFi Signal"
    update_interval: 60s
    entity_category: diagnostic
  - platform: homeassistant
    id: outside_temperature
    entity_id: sensor.wetterstation_terasse_temperature
    internal: true
  - platform: dht
    model: DHT22
    pin: GPIO2
    temperature:
      name: "Room2 Temperature"
      id: room2_temperature
    humidity:
      name: "Room2 Humidity"
      id: room2_humidity
    update_interval: 60s

text_sensor:
  - platform: version
    name: "${friendly_name} ESPHome version"
    hide_timestamp: true
    entity_category: diagnostic
  - platform: wifi_info
    ip_address:
      name: "${friendly_name} IP Address"
      icon: mdi:wifi
      entity_category: diagnostic
    ssid:
      name: "${friendly_name} Connected SSID"
      icon: mdi:wifi-strength-2
      entity_category: diagnostic

binary_sensor:
  - platform: status
    name: "${friendly_name} Status"
    entity_category: diagnostic

graph:
  - id: room2_temperature_graph
    sensor: room2_temperature
    duration: 24h
    x_grid: 4h
    y_grid: 5.0
    width: 128
    height: 64

output:
  - platform: esp8266_pwm
    id: output_red
    pin: GPIO5
  - platform: esp8266_pwm
    id: output_green
    pin: GPIO4
  - platform: esp8266_pwm
    id: output_blue
    pin: GPIO0

light:
  - platform: rgb
    name: "Room2 Light"
    red: output_red
    green: output_green
    blue: output_blue

font:
  - file: 'arial.ttf'
    id: font1
    size: 12
    
  - file: 'arial.ttf'
    id: font2
    size: 34
    
  - file: 'arial.ttf'
    id: font3
    size: 16

image:
  - file: 'sew.png'
    id: sew
    resize: 48x48

i2c:
  sda: GPIO12
  scl: GPIO13
  scan: False

display:
  - platform: ssd1306_i2c
    model: "SSD1306 128x64"
    reset_pin: D0
    address: 0x3C
    rotation: 180°
    id: room2_display
    pages:
      - id: page1
        lambda: |-
          it.printf(60, 0, id(font3), TextAlign::TOP_CENTER, "Naehzimmer");
          it.strftime(0, 50, id(font2), TextAlign::BASELINE_LEFT, "%H:%M", id(esptime).now());

          if (id(outside_temperature).has_state()) {
            it.printf(120, 30, id(font3), TextAlign::TOP_RIGHT, "%.1f°", id(outside_temperature).state);
          }
      - id: page2
        lambda: |-
          it.image(0, 10, id(sew));
          
          if (id(room2_temperature).has_state()) {
            it.printf(62, 15, id(font3), "%.1f°C", id(room2_temperature).state);
          }

          if (id(room2_humidity).has_state()) {
            it.printf(62, 35, id(font3), "%.1f%%", id(room2_humidity).state);
          }

interval:
  - interval: 10s
    then:
      - display.page.show_next: room2_display
      - component.update: room2_display
```


### Anything in the logs that might be useful for us?

```txt
INFO Reading configuration /config/esphome/room2-info.yaml...
INFO Detected timezone 'Europe/Berlin'
INFO Generating C++ source...
INFO Compiling app...
Processing room2-info (board: d1_mini; framework: arduino; platform: platformio/espressif8266 @ 3.2.0)
--------------------------------------------------------------------------------
HARDWARE: ESP8266 80MHz, 80KB RAM, 4MB Flash
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
Dependency Graph
|-- <ESPAsyncTCP-esphome> 1.2.3
|-- <ESPAsyncWebServer-esphome> 2.1.0
|   |-- <ESPAsyncTCP-esphome> 1.2.3
|   |-- <Hash> 1.0
|   |-- <ESP8266WiFi> 1.0
|-- <DNSServer> 1.1.1
|-- <ESP8266WiFi> 1.0
|-- <ESP8266mDNS> 1.2
|-- <Wire> 1.0
|-- <Improv> 1.2.1
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/api/api_connection.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/api/api_frame_helper.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/api/api_pb2.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/api/api_pb2_service.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/api/api_server.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/api/list_entities.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/api/proto.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/api/subscribe_state.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/api/user_services.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/api/util.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/binary_sensor/automation.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/binary_sensor/binary_sensor.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/binary_sensor/filter.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/captive_portal/captive_portal.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/dht/dht.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/display/display_buffer.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/esp8266/core.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/esp8266/gpio.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/esp8266/preferences.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/esp8266_pwm/esp8266_pwm.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/graph/graph.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/homeassistant/sensor/homeassistant_sensor.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/homeassistant/time/homeassistant_time.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/i2c/i2c.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/i2c/i2c_bus_arduino.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/i2c/i2c_bus_esp_idf.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/improv_serial/improv_serial_component.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/light/addressable_light.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/light/automation.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/light/esp_color_correction.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/light/esp_hsv_color.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/light/esp_range_view.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/light/light_call.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/light/light_json_schema.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/light/light_output.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/light/light_state.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/logger/logger.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/md5/md5.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/mdns/mdns_component.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/mdns/mdns_esp32_arduino.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/mdns/mdns_esp8266.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/mdns/mdns_esp_idf.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/network/util.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/ota/ota_backend_arduino_esp32.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/ota/ota_backend_arduino_esp8266.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/ota/ota_backend_esp_idf.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/ota/ota_component.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/output/automation.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/output/float_output.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/sensor/automation.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/sensor/filter.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/sensor/sensor.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/socket/bsd_sockets_impl.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/socket/lwip_raw_tcp_impl.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/socket/socket.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/ssd1306_base/ssd1306_base.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/ssd1306_i2c/ssd1306_i2c.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/status/status_binary_sensor.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/text_sensor/filter.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/text_sensor/text_sensor.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/time/automation.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/time/real_time_clock.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/uptime/uptime_sensor.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/version/version_text_sensor.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/web_server_base/web_server_base.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/wifi/wifi_component.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/wifi/wifi_component_esp32_arduino.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/wifi/wifi_component_esp8266.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/wifi/wifi_component_esp_idf.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/wifi_info/wifi_info_text_sensor.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/components/wifi_signal/wifi_signal_sensor.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/core/application.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/core/color.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/core/component.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/core/controller.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/core/entity_base.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/core/helpers.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/core/log.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/core/scheduler.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/esphome/core/util.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/src/main.cpp.o
Generating LD script /data/room2-info/.pioenvs/room2-info/ld/local.eagle.app.v6.common.ld
Compiling /data/room2-info/.pioenvs/room2-info/lib975/ESPAsyncTCP-esphome/AsyncPrinter.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/lib975/ESPAsyncTCP-esphome/ESPAsyncTCP.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/lib975/ESPAsyncTCP-esphome/ESPAsyncTCPbuffer.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/lib975/ESPAsyncTCP-esphome/SyncClient.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/lib975/ESPAsyncTCP-esphome/tcp_axtls.c.o
Compiling /data/room2-info/.pioenvs/room2-info/libaf0/Hash/Hash.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/lib67b/ESP8266WiFi/BearSSLHelpers.cpp.o
Archiving /data/room2-info/.pioenvs/room2-info/lib975/libESPAsyncTCP-esphome.a
Compiling /data/room2-info/.pioenvs/room2-info/lib67b/ESP8266WiFi/CertStoreBearSSL.cpp.o
Archiving /data/room2-info/.pioenvs/room2-info/libaf0/libHash.a
Compiling /data/room2-info/.pioenvs/room2-info/lib67b/ESP8266WiFi/ESP8266WiFi.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/lib67b/ESP8266WiFi/ESP8266WiFiAP.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/lib67b/ESP8266WiFi/ESP8266WiFiGeneric.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/lib67b/ESP8266WiFi/ESP8266WiFiGratuitous.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/lib67b/ESP8266WiFi/ESP8266WiFiMulti.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/lib67b/ESP8266WiFi/ESP8266WiFiSTA-WPS.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/lib67b/ESP8266WiFi/ESP8266WiFiSTA.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/lib67b/ESP8266WiFi/ESP8266WiFiScan.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/lib67b/ESP8266WiFi/WiFiClient.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/lib67b/ESP8266WiFi/WiFiClientSecureBearSSL.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/lib67b/ESP8266WiFi/WiFiServer.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/lib67b/ESP8266WiFi/WiFiServerSecureBearSSL.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/lib67b/ESP8266WiFi/WiFiUdp.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/lib67b/ESP8266WiFi/enable_wifi_at_boot_time.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/libc74/ESPAsyncWebServer-esphome/AsyncEventSource.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/libc74/ESPAsyncWebServer-esphome/AsyncWebSocket.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/libc74/ESPAsyncWebServer-esphome/WebAuthentication.cpp.o
Archiving /data/room2-info/.pioenvs/room2-info/lib67b/libESP8266WiFi.a
Compiling /data/room2-info/.pioenvs/room2-info/libc74/ESPAsyncWebServer-esphome/WebHandlers.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/libc74/ESPAsyncWebServer-esphome/WebRequest.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/libc74/ESPAsyncWebServer-esphome/WebResponses.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/libc74/ESPAsyncWebServer-esphome/WebServer.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/libecb/DNSServer/DNSServer.cpp.o
Archiving /data/room2-info/.pioenvs/room2-info/libecb/libDNSServer.a
Compiling /data/room2-info/.pioenvs/room2-info/lib3ff/ESP8266mDNS/ESP8266mDNS.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/lib3ff/ESP8266mDNS/LEAmDNS.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/lib3ff/ESP8266mDNS/LEAmDNS_Control.cpp.o
Archiving /data/room2-info/.pioenvs/room2-info/libc74/libESPAsyncWebServer-esphome.a
Compiling /data/room2-info/.pioenvs/room2-info/lib3ff/ESP8266mDNS/LEAmDNS_Helpers.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/lib3ff/ESP8266mDNS/LEAmDNS_Structs.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/lib3ff/ESP8266mDNS/LEAmDNS_Transfer.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/libb5a/Wire/Wire.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/lib251/Improv/improv.cpp.o
Archiving /data/room2-info/.pioenvs/room2-info/libb5a/libWire.a
Archiving /data/room2-info/.pioenvs/room2-info/libFrameworkArduinoVariant.a
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/Crypto.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/Esp-frag.cpp.o
Archiving /data/room2-info/.pioenvs/room2-info/lib251/libImprov.a
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/Esp-version.cpp.o
Archiving /data/room2-info/.pioenvs/room2-info/lib3ff/libESP8266mDNS.a
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/Esp.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/FS.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/FSnoop.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/FunctionalInterrupt.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/HardwareSerial.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/IPAddress.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/LwipDhcpServer-NonOS.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/LwipDhcpServer.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/LwipIntf.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/LwipIntfCB.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/MD5Builder.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/Print.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/Schedule.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/StackThunk.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/Stream.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/StreamSend.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/Tone.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/TypeConversion.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/Updater.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/WMath.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/WString.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/abi.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/aes_unwrap.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/base64.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/cbuf.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/cont.S.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/cont_util.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/core_esp8266_app_entry_noextra4k.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/core_esp8266_eboot_command.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/core_esp8266_features.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/core_esp8266_flash_quirks.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/core_esp8266_flash_utils.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/core_esp8266_i2s.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/core_esp8266_main.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/core_esp8266_non32xfer.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/core_esp8266_noniso.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/core_esp8266_phy.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/core_esp8266_postmortem.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/core_esp8266_si2c.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/core_esp8266_sigma_delta.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/core_esp8266_spi_utils.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/core_esp8266_timer.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/core_esp8266_vm.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/core_esp8266_waveform_phase.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/core_esp8266_waveform_pwm.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/core_esp8266_wiring.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/core_esp8266_wiring_analog.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/core_esp8266_wiring_digital.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/core_esp8266_wiring_pulse.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/core_esp8266_wiring_pwm.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/core_esp8266_wiring_shift.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/crc32.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/debug.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/exc-c-wrapper-handler.S.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/exc-sethandler.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/flash_hal.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/gdb_hooks.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/heap.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/hwdt_app_entry.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/libb64/cdecode.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/libb64/cencode.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/libc_replacements.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/mmu_iram.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/reboot_uart_dwnld.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/spiffs/spiffs_cache.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/spiffs/spiffs_check.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/spiffs/spiffs_gc.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/spiffs/spiffs_hydrogen.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/spiffs/spiffs_nucleus.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/spiffs_api.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/sqrt32.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/stdlib_noniso.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/time.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/uart.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/umm_malloc/umm_info.c.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/umm_malloc/umm_integrity.c.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/umm_malloc/umm_local.c.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/umm_malloc/umm_malloc.cpp.o
Compiling /data/room2-info/.pioenvs/room2-info/FrameworkArduino/umm_malloc/umm_poison.c.o
Archiving /data/room2-info/.pioenvs/room2-info/libFrameworkArduino.a
Linking /data/room2-info/.pioenvs/room2-info/firmware.elf
/data/cache/platformio/packages/toolchain-xtensa/bin/../lib/gcc/xtensa-lx106-elf/10.3.0/../../../../xtensa-lx106-elf/bin/ld: /data/room2-info/.pioenvs/room2-info/firmware.elf section `.text1' will not fit in region `iram1_0_seg'
collect2: error: ld returned 1 exit status
*** [/data/room2-info/.pioenvs/room2-info/firmware.elf] Error 1
========================= [FAILED] Took 327.44 seconds =========================
```


### Additional information

It worked with version 2021.1.3.

Same issue here, 2022.2.1 has the issue, and it was working fine with previous versions 2022.1.4. I wonder if some libraries were changed and now using too much RAM. It's not affecting all of my devices, but most of them (especially ones with OLED that process text).

Here's one that won't compile:
```
substitutions:
  name: "master"
  friendly_name: "Master"

esphome:
  name: ${name}
  platform: ESP8266
  board: nodemcuv2

wifi:
  ssid: !secret wifi_ssid
  password: !secret wifi_password

logger:

api:
  password: !secret master_password
  encryption:
    key: !secret encryption_key
# Enable ota
ota:
  password: !secret ota_password

# Device Specific Configs ###########################################
# Wemos D1 Mini
#
# dallas -> GPIO16 (D0)
# sda -> GPIO5 (D1)
# scl -> GPIO4 (D2)
# soft rx -> GPIO0 (D3) [keep high on boot!!!]
# built in LED -> GPIO2 (D4) [keep high on boot!!!]
# DHT -> GPIO14(D5)
# motion -> GPIO12(D6)
# soft tx -> GPIO13(D7)
# button (to Vcc) -> GPIO15(D8) [keep low on boot!!!]

uart:
  tx_pin: D7
  rx_pin: D3
  baud_rate: 9600

i2c:
  sda: D1
  scl: D2

dfplayer:

time:
  - platform: homeassistant
    id: ha_time
    
text_sensor:
  - platform: homeassistant
    name: "ESPhome Voice"
    internal: true
    entity_id: input_text.esphome_voice
    on_value:
      then:
        - dfplayer.set_volume: 25
        - delay: 150ms
        - dfplayer.play_folder:
            folder: 1
            file: !lambda return atoi(x.c_str());
            
font:
  - file: "verdana.ttf"
    id: my_fontlg
    size: 20
  - file: "verdana.ttf"
    id: my_fontmd
    size: 18
  - file: "verdana.ttf"
    id: my_fontsm
    size: 10
    
display:
  - platform: ssd1306_i2c
    id: my_display
    pages:
      - id: page1
        lambda: |-
          auto nowTime = id(ha_time).now();
          int date = nowTime.day_of_month;
          
          it.strftime(120, 20, id(my_fontlg), TextAlign::BOTTOM_RIGHT, "%I:%M", nowTime);
          if (nowTime.hour > 0 && nowTime.hour < 10) {
            it.filled_rectangle(57, 0, 15, 21, COLOR_OFF);
          }
          else if (nowTime.hour >= 13 && nowTime.hour <= 21)  {
            it.filled_rectangle(57, 0, 15, 21, COLOR_OFF);
          }
          
          if (nowTime.hour < 12) {
            it.print(121, 10, id(my_fontsm), TextAlign::BOTTOM_LEFT, "A");
          }
          else {
            it.print(121, 18, id(my_fontsm), TextAlign::BOTTOM_LEFT, "P");
          }
          
          it.strftime(0, 10, id(my_fontsm), TextAlign::BOTTOM_LEFT, "%A", nowTime);
          
          it.strftime(0, 18, id(my_fontsm), TextAlign::BOTTOM_LEFT, "%b. ", nowTime);
          if (nowTime.day_of_month == 1 || nowTime.day_of_month == 21 || nowTime.day_of_month == 31)  {
            it.printf(49, 18, id(my_fontsm), TextAlign::BOTTOM_RIGHT, "%dst", date);
          }
          else if (nowTime.day_of_month == 2 || nowTime.day_of_month == 22) {
            it.printf(49, 18, id(my_fontsm), TextAlign::BOTTOM_RIGHT, "%dnd", date);
          }
          else if (nowTime.day_of_month == 3 || nowTime.day_of_month == 23) {
            it.printf(49, 18, id(my_fontsm), TextAlign::BOTTOM_RIGHT, "%drd", date);
          }
          else {
            it.printf(49, 18, id(my_fontsm), TextAlign::BOTTOM_RIGHT, "%dth", date);
          }
          
          it.print(it.get_width() / 4, 16, id(my_fontsm), TextAlign::TOP_CENTER, "Inside");
          it.printf(it.get_width() / 4, 24, id(my_fontmd), TextAlign::TOP_CENTER, "%.0f %%", id(thermostat_humid).state);
          it.printf(it.get_width() / 4, 42, id(my_fontlg), TextAlign::TOP_CENTER, "%.0f F", id(inside_temp).state);
          it.print(3*it.get_width() / 4, 16, id(my_fontsm), TextAlign::TOP_CENTER, "Outside");
          it.printf(3*it.get_width() / 4, 24, id(my_fontmd), TextAlign::TOP_CENTER, "%.0f %%", id(backyard_humid).state);
          it.printf(3*it.get_width() / 4, 42, id(my_fontlg), TextAlign::TOP_CENTER, "%.0f F", id(backyard_temp).state);
          
          it.line(0, 16, 127, 16);
          it.line(63, 16, 63, 63);
          
          if (id(button).state) {
            it.rectangle(0, 0, 128, 64);
          }
    model: "SSD1306 128x64"
    reset_pin: D0
    address: 0x3C
interval:
  - interval: 3s
    then:
      - display.page.show_next: my_display
      - component.update: my_display

# Binaries
binary_sensor:
  - platform: status
    name: "${friendly_name} Status"
    
  - platform: gpio
    pin:
      number: GPIO15
      mode: INPUT_PULLUP
    name: "${friendly_name} Button"
    id: button
    filters:
      - delayed_on: 10ms
      - delayed_off: 10ms
    
  - platform: gpio
    pin:
      number: D6
      mode: INPUT
    name: "${friendly_name} Motion"
    device_class: motion
    filters:
      - delayed_off: 5000ms
  
dallas:
  - pin: D0
  
status_led:
  pin: D4

sensor:
  - platform: dallas
    index: 0
    resolution: 12
    accuracy_decimals: 1
    name: "${friendly_name} Temperature"
    id: celsius
    unit_of_measurement: "°F"
    filters:
      - lambda: return x * (9.0 / 5.0) + 32.0;
      
  - platform: wifi_signal
    name: "${friendly_name} WiFi Signal"
    update_interval: 60s
    
  - platform: dht
    pin: D5
    model: DHT22
    humidity:
      accuracy_decimals: 0
      name: "${friendly_name} Humidity"
    update_interval: 60s
        
  - platform: bh1750
    name: "${friendly_name} Luminance"
    address: 0x23
    update_interval: 10s
        
  - platform: homeassistant
    name: "Backyard Temp"
    id: backyard_temp
    entity_id: sensor.backyard_temp
    internal: true
  - platform: homeassistant
    name: "Backyard Humid"
    id: backyard_humid
    entity_id: sensor.backyard_humid
    internal: true
  - platform: homeassistant
    name: "Inside Temp"
    id: inside_temp
    entity_id: sensor.inside_temp_avg
    internal: true
  - platform: homeassistant
    name: "Inside Humid"
    id: thermostat_humid
    entity_id: sensor.thermostat_humidity
    internal: true
```

...and here is the error out:

```
Linking /data/master/.pioenvs/master/firmware.elf
/data/cache/platformio/packages/toolchain-xtensa/bin/../lib/gcc/xtensa-lx106-elf/10.3.0/../../../../xtensa-lx106-elf/bin/ld: /data/master/.pioenvs/master/firmware.elf section `.text1' will not fit in region `iram1_0_seg'
collect2: error: ld returned 1 exit status
*** [/data/master/.pioenvs/master/firmware.elf] Error 1
========================= [FAILED] Took 16.18 seconds =========================
```

On a side note, I noticed compiling seems to take a lot longer since this update, even for the devices that are still working. Not sure if it's related, but thought I'd mention the observation.

--- a/esphome/core/scheduler.cpp
+++ b/esphome/core/scheduler.cpp
@@ -116,7 +116,7 @@ optional<uint32_t> HOT Scheduler::next_schedule_in() {
     return 0;
   return next_time - now;
 }
-void IRAM_ATTR HOT Scheduler::call() {
+void HOT Scheduler::call() {
   const uint32_t now = this->millis_();
   this->process_to_add();
 

